core/src/main/java/it/pagopa/selfcare/mscore/core/UserEventServiceImpl.java

```diff
@@ -79,19 +82,45 @@ public void sendLegalTokenUserNotification(Token token) {
             List<UserToNotify> usersToNotify = toUserToNotify(tokenUser.getUserId(), token.getInstitutionId(), token.getProductId(), Optional.empty(), Optional.of(token.getId()));
             usersToNotify.forEach(user -> {
                 UserNotificationToSend notification = notificationMapper.setNotificationDetailsFromToken(token, user, QueueEvent.ADD);
+                String id = user.getUserId().concat("_"+notification.getInstitutionId()).concat("_"+notification.getProductId()).concat("_"+user.getProductRole());
```

Comment:
it would be better to use a method to construct this id, probably using a string builder would be more readable